dockerfiles/Dockerfile.lint

```diff
@@ -1,23 +1,18 @@
 # syntax=docker/dockerfile:1.3
 
 ARG GO_VERSION=1.13.15
-ARG GOLANGCI_LINTER_SHA="v1.21.0"
+ARG GOLANGCI_LINT_VERSION=v1.21.0
 
-FROM    golang:${GO_VERSION}-alpine AS build
-ENV     CGO_ENABLED=0
-RUN     apk add --no-cache git
-ARG     GOLANGCI_LINTER_SHA
-ARG     GO111MODULE=on
-RUN --mount=type=cache,target=/root/.cache/go-build \
-    --mount=type=cache,target=/go/pkg/mod \
-        go get github.com/golangci/golangci-lint/cmd/golangci-lint@${GOLANGCI_LINTER_SHA}
-
-FROM    golang:${GO_VERSION}-alpine AS lint
-ENV     CGO_ENABLED=0
-ENV     DISABLE_WARN_OUTSIDE_CONTAINER=1
-COPY --from=build /go/bin/golangci-lint /usr/local/bin
+FROM golang:${GO_VERSION}-alpine AS base
 WORKDIR /go/src/github.com/docker/cli
-ENV     GOGC=75
-ENTRYPOINT ["/usr/local/bin/golangci-lint"]
-CMD     ["run", "--config=.golangci.yml"]
-COPY    . .
+
+FROM golangci/golangci-lint:${GOLANGCI_LINT_VERSION}-alpine AS golangci-lint
+
+FROM base AS lint
+ENV CGO_ENABLED=0
+ENV GOGC=75
+RUN --mount=type=bind,target=. \
+  --mount=type=cache,target=/root/.cache/go-build \
+  --mount=type=cache,target=/root/.cache/golangci-lint \
+  --mount=from=golangci-lint,source=/usr/bin/golangci-lint,target=/usr/bin/golangci-lint \
```

Comment:
Yes `RUN --mount` is more efficient because `COPY` implies an uneeded FileOp.

Reply:
But the `COPY` will be once (and after it's done, the intermediate stage will no longer be needed), but with a `--mount`, the intermediate stage cannot be deleted, because it's still needed to mount from, correct?

Reply:
As discussed on Slack I've moved to a `COPY` instruction. :+1: